browser/ui/views/window_closing_confirm_dialog_view.cc

```diff
@@ -0,0 +1,204 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/window_closing_confirm_dialog_view.h"
+
+#include <utility>
+
+#include "base/bind.h"
+#include "base/callback.h"
+#include "base/strings/string_number_conversions.h"
+#include "brave/common/pref_names.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "build/build_config.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_window.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "chrome/grit/chromium_strings.h"
+#include "components/constrained_window/constrained_window_views.h"
+#include "components/prefs/pref_service.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/gfx/font_list.h"
+#include "ui/gfx/geometry/insets.h"
+#include "ui/views/controls/button/checkbox.h"
+#include "ui/views/controls/label.h"
+#include "ui/views/controls/styled_label.h"
+#include "ui/views/layout/box_layout.h"
+#include "ui/views/widget/widget.h"
+#include "ui/views/window/dialog_delegate.h"
+
+namespace {
+
+base::RepeatingCallback<void(Browser*, views::DialogDelegateView*)>
+    g_creation_callback_for_testing;
```

Comment:
pls use NoDestructor and a function for this, we don't want to run constructors for such things at the very start.

Reply:
`NoDestructor` should be placed inside a function. With the current code we still trigger the constructor at the very start.
This is what should be here
```
base::RepeatingCallback<void(views::DialogDelegateView*)>& GetCreationCallbackForTesting() {
  base::NoDestructor<...> callback;
  return *callback;
}
```

Reply:
Done. Thanks!